src/qml/filters/audio_pitch/ui.qml

```diff
@@ -0,0 +1,201 @@
+            }
+        } else if (!octaveKeyframesButton.checked) {
+            filter.resetProperty('octaveshift')
+            filter.set('octaveshift', middleValue)
+        } else if (position !== null) {
+            filter.set('octaveshift', octaveSlider.value, position)
+        }
+    }
+
+    GridLayout {
+        columns: 4
+        anchors.fill: parent
+        anchors.margins: 8
+
+        Label {
+            text: qsTr('Preset')
+            Layout.alignment: Qt.AlignRight
+        }
+        Preset {
+            id: preset
+            Layout.columnSpan: parent.columns - 1
+            parameters: ['octaveshift']
+            onBeforePresetLoaded: {
+                filter.resetProperty(parameters[0])
+            }
+            onPresetSelected: {
+                setControls()
+                octaveKeyframesButton.checked = filter.keyframeCount(parameters[0]) > 0 && filter.animateIn <= 0 && filter.animateOut <= 0
+                middleValue = filter.getDouble(parameters[0], filter.animateIn)
+                if (filter.animateIn > 0)
+                    startValue = filter.getDouble(parameters[0], 0)
+                if (filter.animateOut > 0)
+                    endValue = filter.getDouble(parameters[0], filter.duration - 1)
+            }
+        }
+
+        Label {
+            text: qsTr('Octave Shift')
+            Layout.alignment: Qt.AlignRight
+            ToolTip { text: qsTr('Specify the pitch shift in octaves. -1 shifts down an octave. +1 shifts up an octave. 0 is unchanged.') }
+        }
+        SliderSpinner {
+            id: octaveSlider
+            minimumValue: -1.0
+            maximumValue: 1.0
+            decimals: 3
+            onValueChanged: {
+                updateFilter(getPosition())
+                if (frequencySlider.noUpdate == false) {
+                    frequencySlider.value = Math.pow(2, value)
+                }
+            }
+        }
+        UndoButton {
+            onClicked: octaveSlider.value = 0.0
+        }
+        KeyframesButton {
+            id: octaveKeyframesButton
+            checked: filter.animateIn <= 0 && filter.animateOut <= 0 && filter.keyframeCount('octaveshift') > 0
+            onToggled: {
+                if (checked) {
+                    blockUpdate = true
+                    filter.clearSimpleAnimation('octaveshift')
+                    blockUpdate = false
+                    filter.set('octaveshift', octaveSlider.value, getPosition())
+                } else {
+                    filter.resetProperty('octaveshift')
+                    filter.set('octaveshift', octaveSlider.value)
+                }
+            }
+        }
+
+        Label {
+            text: qsTr('Frequency Scale')
+            Layout.alignment: Qt.AlignRight
+            ToolTip { text: qsTr('Specify the pitch shift as a frequency scale. 0.5 halves the frequency (down one octave). 2 doubles the frequency (up one octave). 1 is unchanged.') }
```

Comment:
Is there a way to express this as the inverse so one can use the same value from Speed here?

Reply:
That is an interesting idea that I did not think of. My plan was to add a "pitch compensation" button to the avformat producer widget to handle the case of pitch correction for speed.

I am very open to discussion about the representation of the ratio. Options:
1) Octaves (currently implemented)
2) Frequency scale (currently implemented)
3) Semitones
4) Speed compensation ratio
5) maybe something else in musical notation?

Reply:
I figured you would do that, but I plan to make beta release this Sunday. I expect to include this PR, but I do not know if you can get Speed integration done. If not, a simple way to simply copy the value from Speed to this filter would be ideal since I think 90% of the uses will be for that. Otherwise, I do not fell more representations are needed at this time.

Reply:
Speed compensation by Sunday would be tight...

Following on your suggestion... How about if I replace "Frequency Scale" with "Speed Compensation" and it will have a range 0.25 - 4.0. But it will apply the inverse of that value to match the corresponding speed. That would be easy to implement.

Reply:
Yes, please

Reply:
Done. See the latest. Also doubled the range.

The idea works well. Type in a speed change in the properties panel, add a pitch filter, type the same number into the pitch compensation field and the audio sounds normal.